src/lgtm.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  addLabels,
+  getPrApprovalNumber,
+  removeLabel,
+  setCommitStatus,
+} from "./github.ts";
+
+// given a pr number, set its lgtm status check and lgtm label
+export const setPrStatusAndLabel = async (
+  pr: {
+    labels: { name: string }[];
+    head: { sha: string };
+    title: string;
+    number: number;
+  },
+) => {
+  let approvals;
+  try {
+    approvals = await getPrApprovalNumber(pr.number);
+  } catch (error) {
+    console.error(error);
+    return;
+  }
+
+  const { state, message, desiredLabel } = getPrStatusAndLabel(approvals);
+  const currentLgtmLabels = pr.labels.filter((l) => l.name.startsWith("lgtm/"));
+
+  // remove any undesired lgtm labels
+  await Promise.all(
+    currentLgtmLabels.filter((l) => l.name !== desiredLabel).map(
+      async (label) => {
+        const response = await removeLabel(pr.number, label.name);
+        if (response.ok) {
+          console.info(
+            `Removed ${label.name} from "${pr.title}" (#${pr.number})`,
+          );
+        } else {
+          console.error(
+            `Failed to remove ${label.name} from "${pr.title}" (#${pr.number})`,
+          );
+          console.error(await response.text());
+        }
+      },
+    ),
+  );
+
+  // add desired label if it's not there
+  if (!currentLgtmLabels.some((label) => label.name === desiredLabel)) {
+    await addLabels(pr.number, [desiredLabel]);
+  }
+
+  // set commit status
+  const response = await setCommitStatus(pr.head.sha, state, message);
+  if (response.ok) {
+    console.info(
+      `Set commit status in "${pr.title}" (#${pr.number})`,
+    );
+  } else {
+    console.error(
+      `Failed to set commit status in  "${pr.title}" (#${pr.number})`,
+    );
+    console.error(await response.text());
+  }
+};
+
+// returns the status, message, and label for a given number of approvals
+export const getPrStatusAndLabel = (approvals: number) => {
+  let desiredLabel = "lgtm/need 2";
```

Comment:
Maybe make this `2` a number constant and make the code below use it? Would be nice, in case we ever want to change this number to for example 3.

Reply:
That's quite a bit of logic (message templating) for something unlikely... If we want it to be 3 it's not a lot of effort (as opposed to the general case)

Reply:
For this function alone, it should be little work, but I guess you mean there's stuff outside the function that needs the number?

Reply:
I mean the message variable (uses two instead of 2)

Reply:
Ah, yes the "2" to "two" convertion.